Document that we conform to SPEC0 version support. (#487)
UCL-ARC/python-tooling: CONTRIBUTING.md

You can omit the `--checkout` option if you're already on the
branch you want to test.

#### Python version support

[![SPEC 0 — Minimum Supported Dependencies](https://img.shields.io/badge/SPEC-0-green?labelColor=%23004811&color=%235CA038)](https://scientific-python.org/specs/spec-0000/)

> We follow Scientific Python's [SPEC0](https://scientific-python.org/specs/spec-0000/) policy.

Reviewer: ```suggestion
We follow Scientific Python's [SPEC 0](https://scientific-python.org/specs/spec-0000/) policy.
```

> Notably this means our template will support [Python versions for three years after their release](https://scientific-python.org/specs/spec-0000/#support-window), and changes to the template are equired to pass tests in all such python versions.

Author: ```suggestion
Notably this means our template will support [Python versions for three years after their release](https://scientific-python.org/specs/spec-0000/#support-window), and changes to the template are required to pass tests in all such python versions.
```

> Notably this means our template will support [Python versions for three years after their release](https://scientific-python.org/specs/spec-0000/#support-window), and changes to the template are equired to pass tests in all such python versions.

Reviewer: ```suggestion
Notably this means our template will support [Python versions for three years after their release](https://scientific-python.org/specs/spec-0000/#support-window), and changes to the template are required to pass tests on all such python versions.
```
